packages/engine/lib/memory/src/shared_memory/segment.rs

```diff
@@ -82,17 +82,29 @@ impl<'id> MemoryId<'id> {
     }
 
     /// Clean up generated shared memory segments associated with a given `MemoryId`.
-    pub fn clean_up<Id: Borrow<Uuid>>(id: Id) {
+    pub fn clean_up<Id: Borrow<Uuid>>(id: Id) -> Result<()> {
         // TODO: macOS does not store the shared memory FDs at `/dev/shm/`. Maybe it's not storing
         //   FDs at all. Find out if they are stored somewhere and remove them instead, otherwise we
         //   have to figure out a way to remove them without relying on the file-system.
-        let shm_files = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id)));
-
-        shm_files.into_iter().flatten().flatten().for_each(|path| {
-            if let Err(err) = std::fs::remove_file(&path) {
-                tracing::warn!("Could not clean up {path:?}: {err}");
-            }
-        });
+        let shm_files = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id)))
+            .map_err(|e| Error::Unique(format!("cleanup glob error: {}", e)))?;
+
+        shm_files
+            .filter_map(|glob_res| match glob_res {
+                Ok(path) => Some(path),
+                Err(err) => {
+                    tracing::warn!(
+                        "Glob Error while trying to clean-up shared memory files: {err}"
+                    );
+                    None
+                }
+            })
+            .for_each(|path| {
+                if let Err(err) = std::fs::remove_file(&path) {
+                    tracing::warn!("Could not clean up {path:?}: {err}");
+                }
+            });
```

Comment:
We don't use any value returned from `for_each`. Can we convert this to a `for` loop instead for better readability?

Reply:
`for_each` doesn't return any values?

Reply:
No, never 😆 
IMO `for_each` is great for chaining another function, but converting this into a for loop would be more readable. No strong opinion, though.

Reply:
What about
```suggestion
        for path in shm_files {
            if let Err(err) = path
                .map_err(GlobError::into_error)
                .and_then(std::fs::remove_file)
            {
                tracing::warn!("Could not remove shared memory file: {err}");
            }
        }
```

Reply:
I couldn't apply the suggestion anymore so https://github.com/hashintel/hash/pull/494/commits/cd3b9ccd8aef181904984b787bda679af5943f0c